lib/Targets/AIERTX.cpp

```diff
@@ -0,0 +1,457 @@
+//===- AIERTX.cpp -----------------------------------------------*- C++ -*-===//
+//
+// Copyright (C) 2023, Advanced Micro Devices, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "aie/Targets/AIERTX.h"
+
+#include "mlir/Support/LogicalResult.h"
+
+extern "C" {
+#include "xaiengine/xaie_core.h"
+#include "xaiengine/xaie_dma.h"
+#include "xaiengine/xaie_interrupt.h"
+#include "xaiengine/xaie_locks.h"
+#include "xaiengine/xaie_plif.h"
+#include "xaiengine/xaie_ss.h"
+#include "xaiengine/xaiegbl.h"
+#include "xaiengine/xaiegbl_defs.h"
+}
+
+using namespace mlir;
+
+#define DEBUG_TYPE "aie-aiertx"
+
+llvm::raw_ostream &operator<<(llvm::raw_ostream &os, const XAie_LocType &loc) {
+  os << "XAie_LocType(col: " << std::to_string(loc.Col)
```

Comment:
I am curious about the rationale of all these `std::to_string`.

Reply:
These are all `uint8_t` which `llvm::raw_ostream` doesn't like for some reason (or something like that) - basically just `<<` didn't do the right thing and I didn't think too hard about it :shrug:

Reply:
Yea that was it - lots of wingdings